prime-router/src/main/kotlin/ReportStreamFilterDefinition.kt

```diff
@@ -370,4 +373,79 @@ class AtLeastOneHasValue : ReportStreamFilterDefinition {
         }
         return selection
     }
+}
+
+/**
+ * InDateInterval(elementName, date, period)
+ *
+ * Implements a test that the specified element (e.g. table column) has a date value between or equal to the
+ * `date` and less than the `date + period`. In other words,
+ * ```
+ * // if period has a positive value
+ * date <= value < date + period
+ * // or if period has a negative value
+ * date + period <= value < date
+ * ```
+ *
+ * ## Examples
+ * ```
+ *   // Specimens collected in 2020
+ *   inDateInterval(specimen_collection_date_time, 202001010000-0000, P1Y)
+ *
+ *   // Specimens collected after 2022-01-01
+ *   inDateInterval(specimen_collection_date_time, 202001010000-0000, P1000Y)
+ *
+ *   // Specimens collected in the past 19 days
+ *   inDateInterval(specimen_collection_date_time, now, -P19D)
+ * ```
+ *
+ * ## Notes
+ *
+ * * There always must be 3 arguments.
+ *
+ * * Empty values in the table never match the interval
+ *
+ * * A `date` argument can be `now` to indicate the current time. Otherwise, it must follow
+ * one of the formats that ReportStream understands. The HL7 date-time format is recommended.
+ *
+ * * A `duration` argument must follow the ISO 8061 standard format as implemented by Java.
+ * This format is `[-]P[n]Y[n]M[n]D` or `P[n]W`.
+ * The period can be negative to indicate an interval that starts before the `date` argument.
+ * The period can be zero but that will never match any value.
+ */
+class InDateInterval : ReportStreamFilterDefinition {
```

Comment:
It looks like there isn't a straightforward way to enforce calling this only on DATE or DATETIME field types.   We should be passing the Report into getSelection, instead of the Table, so that getSelection has access to richer information, such as Element datatypes.

Reply:
Added issue for this follow-on work: #5144